app/util/date_util.py

```diff
@@ -0,0 +1,30 @@
+from datetime import datetime, timedelta
+
+from werkzeug.exceptions import BadRequest
+
+
+def formulate_date(date: str):
+    try:
+        date_arr = date.split("-")
+        if len(date_arr) != 3:
+            raise BadRequest("Date is not properly formatted!")
+        yyyy, mm, dd = [int(i) for i in date_arr]
+        return datetime(yyyy, mm, dd, 0, 0)
+    except:
+        raise BadRequest("Date is not properly formatted!")
+
+
+def get_days_between(date1: datetime, date2: datetime):
+    try:
+        diff = date2 - date1
+        if diff.days < 0:
+            raise Exception()
```

Comment:
Why do we need to wrap this in a try catch when we can raise BadRequest here? It seems that the flow now is for it to check diff.days -> go to catch block then raise another exception?